packages/access-client/src/agent.js

```diff
@@ -409,6 +481,101 @@ export class Agent {
     }
   }
 
+  /**
+   * @param {Ucanto.DID<'key'>} space
+   * @param {Ucanto.Principal<Ucanto.DID<'mailto'>>} account
+   * @param {Ucanto.DID<'web'>} provider - e.g. 'did:web:staging.web3.storage'
+   */
+  async #addProvider(space, account, provider) {
+    return addProvider(this, space, account, provider)
+  }
+
+  /**
+   *
+   * @param {Ucanto.DID<'key'>} space
+   * @param {Ucanto.Principal<Ucanto.DID<'mailto'>>} account
+   */
+  async #delegateSpaceAccessToAccount(space, account) {
+    const issuerSaysAccountCanAdminSpace =
+      await createIssuerSaysAccountCanAdminSpace(
+        this.issuer,
+        space,
+        account,
+        undefined,
+        this.proofs([{ with: space, can: '*' }])
+      )
+    return this.invokeAndExecute(Access.delegate, {
+      audience: this.connection.id,
+      with: space,
+      expiration: Infinity,
+      nb: {
+        delegations: {
+          [issuerSaysAccountCanAdminSpace.cid.toString()]:
+            issuerSaysAccountCanAdminSpace.cid,
+        },
+      },
+      proofs: [
+        await createIssuerSaysDeviceCanAccessDelegateWithSpace(
+          this.issuer,
+          space,
+          this.issuer
```

Comment:
This delegation makes no sense. Issuer delegates access to space to itself. If it already has access it does not need this delegation. If it does not have access than it should have no authority over space to be able to delegate it in first place.

I suspect all of this only because the space and the issuer are the same so this proof is never utilized.